tests/test_routes.py

```diff
@@ -367,3 +367,19 @@ def test_activate_customer_not_found(self):
         # activate the customer
         response = self.client.put(f"{BASE_URL}/0/activate")
         self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        
+    def test_query_customer_list_by_name(self):
+        """It should query customers by name"""
+        customers = self._create_customers(10)
+        test_name = customers[0].name
+        name_customers = [customer for customer in customers if customer.name == test_name]
+        response = self.client.get(
+            BASE_URL,
+            query_string=f"name={quote_plus(test_name)}"
```

Comment:
in order to use the `quote_plus` function, you will have to import it. add the following import below the unittest import on ln 11:

`from urllib.parse import quote_plus`

additionally, you will have to change the query string so it is `first_name` instead of `name`:
`query_string=f"first_name={quote_plus(test_name)}"`